dpub/core.py

```diff
@@ -44,24 +55,46 @@ def run(read_dimension=drive.COLS_DIMENSION,
 
     d = drive.Drive(args.credentials, args.token)
     items = _compose_items()
-    tests_map = _read_tests_map(
+    tests_map, empty_ids = _read_tests_map(
         d, doc, first_test_location, first_msg_location, read_dimension, write_dimension)
 
-    # Append the items to the tests. If having one profile there will be one
-    # trace per profile, but having several, there will be several items
+   # Append the items to the tests. If having one profile there will be one
+   # trace per profile, but having several, there will be several items
     for it in items:
         if not it.test_id in tests_map:
             # TODO SEE WHAT TO DO HERE WITH THE ERROR MESSAGE
-            continue
+            tests_map_len = int(len(tests_map))
+            # It calculates in which row position write a new test case
+            new_item_row = tests_map_len + empty_ids
+            tests_map = _append_tests(
+                tests_map, it.test_id, new_item_row, first_msg_location, read_dimension)
 
         t = tests_map[it.test_id]
         t.append(it)
 
-    # For every test, write the related traces
+# For every test, write the related traces
     for _, t in tests_map.items():
-        m_range = t.first_message_range
-        values = output.compose(t, mode)
-        _write_messages(d, doc, values, m_range, write_dimension)
+        if t.update == "Yes":
+            m_range = t.first_message_range
+            values = output.compose(t, mode)
+            _write_messages(d, doc, values, m_range, write_dimension)
+        elif t.update == "No":
+            continue
+        elif t.update == "New":
+            # TODO: this small part would be moved to a function.
+            # It calculates the letter of the previous column of the requests: the id column letter.
+            # Then, for new cases we can write the test case id as well.
+            m_range = t.first_message_range
```

Comment:
as said above, please move this to a function

Reply:
Done.